hera_cal/smooth_cal.py

```diff
@@ -960,7 +977,7 @@ def time_freq_2D_filter(self, freq_scale=10.0, time_scale=1800.0, tol=1e-09, fil
 
                 # If the weights grid is the same as the previous, the solution matrix can be reused to speed up computation
                 if method == 'DPSS' or method == 'dpss_leastsq':
-                    XTXinv = info['XTXinv'] if np.allclose(wgts_grid, wgts_old) else None
+                    cached_input = info['cached_input'] if np.allclose(wgts_grid, wgts_old) else {}
```

Comment:
This reminds me... I think we might get a speed-up if we sort the gain_grids by the number of flags... so that antennas with the same flagging pattern are more likely to come in a row.

Reply:
Just added a couple lines to sort the antennas by number of flags